frappe/integrations/doctype/webhook/__init__.py

```diff
@@ -4,48 +4,50 @@
 import frappe
 
 
+def get_all_webhooks():
+	# query webhooks
+	webhooks_list = frappe.get_all(
+		"Webhook",
+		fields=["name", "condition", "webhook_docevent", "webhook_doctype"],
+		filters={"enabled": True},
+	)
+
+	# make webhooks map
+	webhooks = {}
+	for w in webhooks_list:
+		webhooks.setdefault(w.webhook_doctype, []).append(w)
```

Comment:
I see the use of `setdefault` on dictionaries all over the frappe codebase, it might also just be a micro optimization, but why not use `defaultdict` collection instead? 
This way we can set the default type on the whole dictionary instead of key, by key.

Reply:
This could be one of the reasons:

```py

In [11]: %timeit pickle.dumps(x)
2.1 µs ± 15.7 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [12]: x
Out[12]: defaultdict(list, {'abc': []})

In [13]: y = dict(x)

In [14]: y
Out[14]: {'abc': []}

In [15]: %timeit pickle.dumps(y)
312 ns ± 2.81 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [16]: z = pickle.dumps(x)

In [17]: %timeit pickle.loads(z)
1.58 µs ± 6.13 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)

In [18]: a = pickle.dumps(y)

In [19]: %timeit pickle.loads(a)
271 ns ± 0.554 ns per loop (mean ± std. dev. of 7 runs, 1,000,000 loops each)
```

essentially, `defaultdict` isn't as optimised for pickling.